### PR TITLE
configure pyupgrade via pre-commit

### PR DESCRIPTION
**Pull request by graingert**
also black was missing some files, eg `setup.py` and `docs/conf.py`

I'd recommend configuring https://pre-commit.ci/

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,3 +1,3 @@
 from setuptools import setup  # type: ignore[import]
 
-setup(name='asgiref')
+setup(name="asgiref")
```
Comment (graingert) [inline setup.py:3]: @andrewgodwin why is this here? it should work with just
```suggestion
setup()
```

Comment (andrewgodwin) [reply setup.py:3]: Presumably it was a copy-and-paste inheritance from an older setup.py. I think we can remove it.

Comment (graingert) [reply setup.py:3]: @andrewgodwin want me to land it in this PR?

Comment (andrewgodwin) [reply setup.py:3]: Sure

**Review by andrewgodwin — APPROVED** [review]
